src/cmd/embeddedrules/rules.php

```diff
@@ -238,3 +238,52 @@ function argsOrder() {
    */
   explode($_, ${"char"}, ${"*"});
 }
+
+/**
+ * @comment Report not-strict-enough comparisons.
+ * @before  in_array("what", $s)
+ * @after   in_array("what", $s, true)
+ */
+function strictCmp() {
+    /**
+     * @warning non-strict comparison (use ===)
+     * @type string $x
+     * @type string $y
+     */
+    any_equal: {
+        $_ == true;
+        true == $_;
+        $_ == false;
+        false == $_;
+        $_ == null;
+        null == $_;
+        $x == $y;
+    }
+
+    /**
+     * @warning non-strict comparison (use !==)
+     * @type string $x
+     * @type string $y
```

Comment:
I'm not sure what these type filters are supposed to do.
Maybe we can remove them?
Patterns below don't have `$x` or `$y` matcher variables.
I would guess it's a copy/paste error.